iml-services/iml-snapshot/src/main.rs

```diff
@@ -27,23 +138,66 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
     let pool = get_db_pool(get_pool_limit().unwrap_or(DEFAULT_POOL_LIMIT)).await?;
     sqlx::migrate!("../../migrations").run(&pool).await?;
 
+    let snapshot_client_counts: Arc<Mutex<HashMap<SnapshotId, Option<State>>>> =
+        Arc::new(Mutex::new(HashMap::new()));
+    let snapshot_client_counts_2 = snapshot_client_counts.clone();
+
+    tokio::spawn(async move {
+        // Avoid conflict with futures_util::StreamExt
+        use tokio::stream::StreamExt;
+        let mut interval = interval(Duration::from_secs(10));
+
+        while let Some(_) = interval.next().await {
+            let tick_result = tick(snapshot_client_counts_2.clone()).await;
+            if let Err(e) = tick_result {
+                tracing::error!("Error during handling snapshot autounmount: {}", e);
+            }
+        }
+    });
+
     while let Some((fqdn, snapshots)) = s.try_next().await? {
         tracing::debug!("snapshots from {}: {:?}", fqdn, snapshots);
 
-        let snaps = snapshots.into_iter().fold(
-            (vec![], vec![], vec![], vec![], vec![], vec![], vec![]),
-            |mut acc, s| {
-                acc.0.push(s.filesystem_name);
-                acc.1.push(s.snapshot_name);
-                acc.2.push(s.create_time.naive_utc());
-                acc.3.push(s.modify_time.naive_utc());
-                acc.4.push(s.snapshot_fsname.clone());
-                acc.5.push(s.mounted);
-                acc.6.push(s.comment);
-
-                acc
-            },
-        );
+        let snaps = {
+            let mut snapshot_client_counts = snapshot_client_counts.lock().await;
```

Comment:
Instead of reading incoming data, we should instead query the table each tick.

This will ensure we start with the same state as the table on service restart instead of having to wait for the MGS to report back in again.

In addition, it means we can keep all the logic inside of `tick` besides initializing the `snapshot_client_counts`, instead of having some of it in here.

Reply:
Done